src/ResourceManager/Sql/Commands.Sql.Test/ScenarioTests/ManagedInstanceKeyVaultKeyTests.ps1

```diff
@@ -0,0 +1,56 @@
+﻿# ----------------------------------------------------------------------------------
+#
+# Copyright Microsoft Corporation
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ----------------------------------------------------------------------------------
+		
+$mangedInstanceRg = "MlAndzic_RG"
+$managedInstanceName = "midemoinstancebc"
+$keyVaultName = "mitest-doNotDelete"
+$keyName = "mitest-key"
+$keyId = "https://mitest-donotdelete.vault.azure.net/keys/mitest-key/e97043455e14493ca05c3642549aedba"
+$keyVersion = "e97043455e14493ca05c3642549aedba"
+$tdeKeyName = $keyVaultName + "_" + $keyName + "_" + $keyVersion
+
+<#
+	.SYNOPSIS
+	Tests adding TDE keyVaultKey to managed instance
+#>
+function Test-AddManagedInstanceKeyVaultKey
+{
+	$keyResult = Add-AzureRmSqlManagedInstanceKeyVaultKey -ResourceGroupName $mangedInstanceRg -ManagedInstanceName $managedInstanceName -KeyId $keyId
```

Comment:
Just combine these three tests together, no need to do setup for each.

Reply:
Also, please add test for remove

Reply:
Currently there is a production bug, which is causing the remove/delete api to fail. It is important to note, that not only this API, but a few other already released APIs are also affected by this bug.
The operation it self completes successfully, but the management service reports failure in post condition check hence, we are unable to validate the remove functionality at the moment.

Reply:
Will combine tests into one